functorch/_src/eager_compilation.py

```diff
@@ -134,14 +135,26 @@ def vjpfull(primals, tangents):
                 out, vjpfn = vjp(flat_fn, *primals)
                 return out, vjpfn(*tangents)
 
+            def outputs_like(outs):
```

Comment:
I think we should probably use pytrees here? I suppose it's possible that we actually don't need to do anything at all... the only reason this is here is to end up initializing the trace with tensors the same shape as the outputs.

Reply:
So... we could just pass the outputs in directly to the trace.